File: LeetCode/128LongestConsecutiveSequence/LongestConsecutiveSequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # HashSet -> Time:O(n), Space:O(n)
        s = set(nums)
        ans = 0
        for i in range(len(nums)):
            curr_seq = 0
            curr_num = nums[i]
            if curr_num - 1 not in s:
                curr_seq = 1
                while curr_num + 1 in s:
                    curr_num += 1
                    curr_seq += 1
            if curr_seq > ans:
                ans = curr_seq
        return ans

File: LeetCode/128LongestConsecutiveSequence/LongestConsecutiveSequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        // HashSet -> Time:O(n), Space:O(n)
        set<int> s(nums.begin(), nums.end());
        int ans = 0, curr_seq, curr_num;
        for (int i = 0; i < nums.size(); i++) {
            curr_num = nums[i];
            curr_seq = 0;
            if (s.find(curr_num-1) == s.end()) {
                curr_seq++;
                while (s.find(curr_num+1) != s.end()) {
                    curr_seq++;
                    curr_num++;
                }
                if (curr_seq > ans) {
                    ans = curr_seq;
                }
            }
        }
        return ans;
    }
};
